common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signup/SignUpStartApiResponse.kt

```diff
@@ -0,0 +1,169 @@
+
+import com.google.gson.annotations.Expose
+import com.google.gson.annotations.SerializedName
+import com.microsoft.identity.common.java.logging.LogSession
+import com.microsoft.identity.common.java.providers.nativeauth.IApiResponse
+import com.microsoft.identity.common.java.util.isAttributeValidationFailed
+import com.microsoft.identity.common.java.util.isAuthNotSupported
+import com.microsoft.identity.common.java.util.isInvalidClient
+import com.microsoft.identity.common.java.util.isInvalidEmail
+import com.microsoft.identity.common.java.util.isInvalidParameter
+import com.microsoft.identity.common.java.util.isPasswordBanned
+import com.microsoft.identity.common.java.util.isPasswordRecentlyUsed
+import com.microsoft.identity.common.java.util.isPasswordTooLong
+import com.microsoft.identity.common.java.util.isPasswordTooShort
+import com.microsoft.identity.common.java.util.isPasswordTooWeak
+import com.microsoft.identity.common.java.util.isRedirect
+import com.microsoft.identity.common.java.util.isUnsupportedChallengeType
+import com.microsoft.identity.common.java.util.isUserAlreadyExists
+import com.microsoft.identity.common.java.util.isVerificationRequired
+import com.microsoft.identity.common.java.util.toAttributeList
+import java.net.HttpURLConnection
+
+/**
+ * Represents the raw response from the Sign Up /start endpoint.
+ * Can be converted to SignUpContinueApiResult using the provided toResult() method.
+ */
+data class SignUpStartApiResponse(
+    @Expose override var statusCode: Int,
+    @Expose @SerializedName("error") val error: String?,
+    @Expose @SerializedName("error_description") val errorDescription: String?,
+    @SerializedName("signup_token") val signupToken: String?,
+    @Expose @SerializedName("unverified_attributes") val unverifiedAttributes: List<Map<String, String>>?,
+    @Expose @SerializedName("invalid_attributes") val invalidAttributes: List<Map<String, String>>?,
+    @SerializedName("details") val details: List<Map<String, String>>?,
+    @Expose @SerializedName("challenge_type") val challengeType: String?,
+    @Expose @SerializedName("error_codes") val errorCodes: List<Int>?
+) : IApiResponse(statusCode) {
+
+    companion object {
+        private val TAG = SignUpStartApiResponse::class.java.simpleName
+    }
+
+    fun toResult(): SignUpStartApiResult {
+        LogSession.logMethodCall(TAG, "${TAG}.toResult")
+
+        return when (statusCode) {
+
+            // Handle 400 errors
+            HttpURLConnection.HTTP_BAD_REQUEST -> {
+                when {
+                    error.isUserAlreadyExists() -> {
+                        SignUpStartApiResult.UsernameAlreadyExists(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    errorCodes?.get(0).isInvalidParameter() and (errorDescription?.isInvalidEmail() == true) -> {
```

Comment:
Why are there multiple error codes? and if there are then why do we only look at index 0?

Reply:
There can be multiple error codes to represent sub error codes. The first error code is the main one and the remaining would be sub error codes.